packages/mobile/src/networkInfo/saga.ts

```diff
@@ -34,6 +74,43 @@ function* subscribeToNetworkStatus() {
   }
 }
 
+function* fetchUserLocationData() {
+  let userLocationData: UserLocationData
+  try {
+    // NOTE: Need to pay for this API if we decideq to use it at scale
+    const response: Response = yield fetchWithTimeout(
+      'https://ipapi.co/json/',
+      null,
+      FETCH_TIMEOUT_DURATION
+    )
+
+    const ipAddressData: IpAddressData = yield response.json()
+
+    if (!response.ok) {
+      throw Error(`IP address fetch failed. Error: ${JSON.stringify(ipAddressData)}`)
```

Comment:
nit: https://stackoverflow.com/questions/38759428/when-is-new-error-better-than-error
```suggestion
      throw new Error(`IP address fetch failed. Error: ${JSON.stringify(ipAddressData)}`)
```